### PR TITLE
Update AWS package to use latest package-spec version

### PR DESCRIPTION
**Pull request by tommyers-elastic**
## What does this PR do?

Update AWS package to use latest package-spec version. This is preparation for serverless.

The required changes here are self-contained within individual commits. Please see the git log for more details. 

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

File: packages/aws/manifest.yml [hunk 1]
```diff
@@ -1,15 +1,15 @@
-format_version: 1.0.0
+format_version: 2.11.0
 name: aws
 title: AWS
-version: 2.3.0
-license: basic
+version: 2.4.0
```
Comment (kaiyan-sheng) [inline packages/aws/manifest.yml:4]: Just curious is this change big enough to bump the version to 3.0.0 by chance?

File: packages/aws/manifest.yml
Comment (taylor-swanson) [inline packages/aws/manifest.yml:1]: We added a new field to package-spec to mark the owner type of the package. It was introduced in 2.11.0 and assumes a default of `community` in 2.11.0, but we decided we didn't like that behavior, so we made it required in 3.0.0. Please add the following:

```diff
 owner:
   github: elastic/obs-cloud-monitoring
+  type: elastic
```

More info on owner type tagging: #6569

**Review by taylor-swanson — CHANGES_REQUESTED** [review]

File: packages/aws/manifest.yml (hunk 1 above)
Comment (tommyers-elastic) [reply packages/aws/manifest.yml:4]: the changes in this PR should result in no behaviour changes at all in the package. i don't think it warrants a major version bump for the package.

File: packages/aws/manifest.yml
Comment (tommyers-elastic) [reply packages/aws/manifest.yml:1]: thanks for the heads up !

**Review by taylor-swanson — APPROVED** [review]

**Review by kaiyan-sheng — APPROVED** [review]

**Review by kaiyan-sheng — COMMENTED** [review]
Based on https://github.com/elastic/package-spec/pull/567 kibana.version is equal or greater to 8.10.0?

**Comment by tommyers-elastic** [comment]
closing since this is being addressed in another PR

**Pull request closed without merging**